api/src/nv_ingest_api/util/dataloader/dataloader.py

```diff
@@ -0,0 +1,177 @@
+# SPDX-FileCopyrightText: Copyright (c) 2025, NVIDIA CORPORATION & AFFILIATES.
+# All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+# Copyright (c) 2025, NVIDIA CORPORATION.
+import fsspec
+from upath import UPath as Path
+from abc import ABC, abstractmethod
+import queue
+import threading
+import subprocess
+import json
+import logging
+import math
+import importlib
+
+logger = logging.getLogger(__name__)
+
+if importlib.util.find_spec("ffmpeg") is None:
+    logger.error(
+        "Unable to load the Dataloader,ffmpeg was not installed, "
+        "please install it using `pip install ffmpeg-python` and `apt-get install ffmpeg`"
+    )
+else:
+    import ffmpeg
+
+
+class LoaderInterface(ABC):
+
+    @abstractmethod
+    def split(self, input_path: str, output_dir: str, split_interval: int = 0):
+        pass
+
+
+def _probe(filename="pipe:", cmd="ffprobe", format=None, file_handle=None, timeout=None, **kwargs):
```

Comment:
Is it possible to do this without calling out to a subprocess? This opens us up to situations where we're more susceptible to deadlocks/hangs that are hard to diagnose, zombie processes, etc..